mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -4351,10 +4435,15 @@ pub async fn create_maker_order(ctx: &MmArc, req: SetPriceReq) -> Result<MakerOr
         rel_coin.coin_protocol_info(),
     );
 
-    {
-        let mut my_maker_orders = ordermatch_ctx.my_maker_orders.lock();
-        my_maker_orders.insert(new_order.uuid, Arc::new(AsyncMutex::new(new_order.clone())));
+    let mut makerorders_ctx = ordermatch_ctx.makerorders_ctx.lock();
+    if !makerorders_ctx.balance_loop_exists(base_coin.ticker()) {
+        let ctx_weak = ctx.weak();
+        let ticker = base_coin.ticker().to_string();
+        spawn(async move { check_balance_update_loop(ctx_weak, ticker, Some(balance)).await });
```

Comment:
Could you please use `common::spawn_abortable` instead, and store the returned `AbortOnDropHandle` in `MakerOrdersContext`? This way, you can remove `if !makerorders_ctx.coin_has_active_maker_orders(&ticker)` check from `check_balance_update_loop` and stop it immediately on `remove_order` by dropping the handle if no more orders left.